src/LowerWarpShuffles.cpp

```diff
@@ -562,6 +562,22 @@ class LowerWarpShuffles : public IRMutator {
 
         internal_assert(may_use_warp_shuffle) << name << ", " << idx << ", " << lane << "\n";
 
+        // We must add .sync after volta architecture:
+        // https://docs.nvidia.com/cuda/volta-tuning-guide/index.html
+        string sync_suffix = "";
+        Target t = get_jit_target_from_environment();
```

Comment:
Didn't notice this before. This is not the right target to use here - Halide is commonly used as a cross compiler and the jit target is usually some sort of host target for the build machine. I think you'll need to change lower_warp_shuffles to also accept a Target object and plumb that through to here.

Reply:
Oh, I see. Fixed now.